done/done.py

```diff
@@ -1,77 +1,110 @@
-"""TO-DO: Show a toggle which lets students mark things as done."""
+""" Show a toggle which lets students mark things as done."""
 
 import pkg_resources
+import uuid
 
 from xblock.core import XBlock
-from xblock.fields import Scope, Integer, String, Boolean, DateTime, Float
+from xblock.fields import Scope, String, Boolean, DateTime, Float
 from xblock.fragment import Fragment
 
+try:
+    from eventtracking import tracker
+except ImportError:
+    class tracker(object):  # pylint: disable=invalid-name
+        """
+        Define tracker if eventtracking cannot be imported. This is a
+        workaround so that the code works in both edx-platform and
+        XBlock workbench (the latter of which does not support event
+        emission). This should be replaced with XBlock's emit(), but
+        at present, emit() is broken.
+        """
+        def __init__(self):
+            """ Do nothing """
+            pass
+
+        @staticmethod
+        def emit(param1, param2):
+            """ In workbench, do nothing for event emission """
+            pass
+
+
+def resource_string(path):
+    """Handy helper for getting resources from our kit."""
+    data = pkg_resources.resource_string(__name__, path)
+    return data.decode("utf8")
+
 
 class DoneXBlock(XBlock):
     """
     Show a toggle which lets students mark things as done.
     """
 
-    # Fields are defined on the class.  You can access them in your code as
-    # self.<fieldname>.
     done = Boolean(
-           scope = Scope.user_state, 
-           help = "Is the student done?",
-           default = False
-        )
+        scope=Scope.user_state,
+        help="Is the student done?",
```

Comment:
Urgh, sorry-- I keep getting confused about i18n for xblocks. Ignore my other i18n comments if not relevant.